sizewatcher fails if there is no "package.json" file in the main branch

### Expected Behaviour
If a new `package.json` gets added, sizewatcher should be able to handle it without crashing.

### Actual Behaviour
sizewatcher crashes.

### Reproduce Scenario (including but not limited to)
- Create a repo and leave the main branch empty.
- Create another branch and add a `package.json` file to it.
- On PR, sizewatcher should fail.

#### Platform and Version
Observed with node 12+.

#### Logs taken while reproducing problem
```
   🚨 node_modules measurement error: ENOENT: no such file or directory, lstat '/tmp/<<REDACTED>>/before/node_modules'

   🚨 npm_package measurement error: Command failed: npm publish --dry-run
npm WARN Local package.json exists, but node_modules missing, did you mean to install?
npm ERR! code ELIFECYCLE
npm ERR! errno 3
npm ERR! @<<REDACTED>>@1.0.0 prepack: <<script>>
npm ERR! Exit status 3
npm ERR!
npm ERR! Failed at the @<<REDACTED>>@1.0.0 prepack script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

```
Effective Configuration:
limits:
  fail: 100%
  warn: 30%
  ok: '-10%'
report:
  githubComment: true
  githubStatus: false
comparators: {}
```

--- a/lib/comparators/node_modules.js
+++ b/lib/comparators/node_modules.js
@@ -24,19 +24,36 @@ function isEmptyObject(obj) {
     return Object.keys(obj).length === 0 && obj.constructor === Object;
 }
 
+function hasPackageJson(dir) {
+    return fs.existsSync(path.join(dir, "package.json"));
+}
+
 async function getSize(dir) {
+    if (!hasPackageJson(dir)) {
+        return 0;
+    }
     process.chdir(dir);
+
     debug(`installing node dependencies inside ${dir}...`);
     if (fs.existsSync("package-lock.json") || fs.existsSync("npm-shrinkwrap.json")) {
         execSync("npm ci");
     } else {
         execSync("npm install");
     }
-    debug(`calculating folder size of ${dir}/node_modules...`);
-    return du(path.join(dir, "node_modules"));
+
+    const nodeModules = path.join(dir, "node_modules");
+    if (fs.existsSync(nodeModules)) {
+        debug(`calculating folder size of ${nodeModules}...`);
+        return du(nodeModules);
+    } else {
+        return 0;
+    }
 }
 
 async function costOfModules(dir) {
+    if (!hasPackageJson(dir)) {
+        return "(no package.json)";
+    }
     process.chdir(dir);
 
     const pkgJsonPath = path.join(dir, "package.json");
@@ -62,7 +79,7 @@ async function costOfModules(dir) {
 module.exports = {
 
     shouldRun: async function(beforeDir, afterDir) {
-        return fs.existsSync(path.join(afterDir, "package.json"));
+        return hasPackageJson(beforeDir) || hasPackageJson(afterDir);
     },
 
     compare: async function(before, after) {

--- a/lib/compare.js
+++ b/lib/compare.js
@@ -136,6 +136,7 @@ function process(deltas) {
     };
 
     for (const d of deltas) {
+        debug("processing delta:", d);
         if (d.error) {
             summary.error++;
             continue;
@@ -147,14 +148,14 @@ function process(deltas) {
         };
 
         let increase;
-        if (d.beforeSize <= 0) {
-            if (d.afterSize <= 0) {
-                increase = 0;
-            } else {
-                increase = -100;
-            }
+        if (d.beforeSize <= 0 && d.afterSize <= 0) {
+            increase = 0;
+        } else if (d.afterSize <= 0) {
+            increase = -100;
+        } else if (d.beforeSize <= 0) {
+            increase = 100;
         } else {
-            increase = d.beforeSize <= 0 ? -100 : (d.afterSize - d.beforeSize) / d.beforeSize * 100;
+            increase = (d.afterSize - d.beforeSize) / d.beforeSize * 100;
         }
         d.increase = increase.toFixed(1);
 

--- a/test/cli.test.js
+++ b/test/cli.test.js
@@ -95,4 +95,43 @@ describe("cli e2e", function() {
         assert(this.output.stdout.includes("git:"));
     });
 
+    it("no package.json in before branch", async function() {
+        await exec(path.join(PROJECT_DIR, "test/scripts/no-package-json-before.sh"));
+
+        await sizewatcher();
+
+        assert.strictEqual(lastExitCode, undefined);
+        assert(!this.output.any.includes("node_modules: measurement error"));
+        assert(this.output.stdout.includes("+ ✅  node_modules:"));
+    });
+
+    it("no package.json in after branch", async function() {
+        await exec(path.join(PROJECT_DIR, "test/scripts/no-package-json-after.sh"));
+
+        await sizewatcher();
+
+        assert.strictEqual(lastExitCode, undefined);
+        assert(!this.output.any.includes("node_modules: measurement error"));
+        assert(this.output.stdout.includes("+ ✅  node_modules:"));
+    });
+
+    it("package.json with dependencies removed", async function() {
+        await exec(path.join(PROJECT_DIR, "test/scripts/package-json-removed.sh"));
+
+        await sizewatcher();
+
+        assert.strictEqual(lastExitCode, undefined);
+        assert(!this.output.any.includes("node_modules: measurement error"));
+        assert(this.output.stdout.includes("+ 🎉  node_modules: -100.0%"));
+    });
+
+    it("package.json with dependencies added", async function() {
+        await exec(path.join(PROJECT_DIR, "test/scripts/package-json-added.sh"));
+
+        await sizewatcher();
+
+        assert.strictEqual(lastExitCode, undefined);
+        assert(!this.output.any.includes("node_modules: measurement error"));
+        assert(this.output.stdout.includes("+ ❌  node_modules: 100.0%"));
+    });
 });

--- a/test/scripts/no-package-json-after.sh
+++ b/test/scripts/no-package-json-after.sh
@@ -1,0 +1,18 @@
+#!/bin/sh
+
+# create a repo where a package.json is added in a branch
+
+# setup new git repo
+git init --initial-branch=main
+# needed for git run in CI context where none of this is set
+git config --local user.email "you@example.com"
+git config --local user.name "Your Name"
+
+echo '{ "name": "test-project" }' > package.json
+git add package.json
+git commit -a -m "add package.json"
+
+# add package.json in new branch
+git checkout -b branch
+git rm package.json
+git commit -a -m "remove package.json"

--- a/test/scripts/no-package-json-before.sh
+++ b/test/scripts/no-package-json-before.sh
@@ -1,0 +1,19 @@
+#!/bin/sh
+
+# create a repo where a package.json is added in a branch
+
+# setup new git repo
+git init --initial-branch=main
+# needed for git run in CI context where none of this is set
+git config --local user.email "you@example.com"
+git config --local user.name "Your Name"
+
+echo "hello" > file
+git add file
+git commit -a -m "initial commit"
+
+# add package.json in new branch
+git checkout -b branch
+echo '{ "name": "test-project" }' > package.json
+git add package.json
+git commit -a -m "add package.json"

--- a/test/scripts/package-json-added.sh
+++ b/test/scripts/package-json-added.sh
@@ -1,0 +1,19 @@
+#!/bin/sh
+
+# create a repo where a package.json is added in a branch
+
+# setup new git repo
+git init --initial-branch=main
+# needed for git run in CI context where none of this is set
+git config --local user.email "you@example.com"
+git config --local user.name "Your Name"
+
+echo "hello" > file
+git add file
+git commit -a -m "initial commit"
+
+# add package.json in new branch
+git checkout -b branch
+echo '{ "name": "test-project", "dependencies": { "debug": "4.3.4"} }' > package.json
+git add package.json
+git commit -a -m "add package.json"

--- a/test/scripts/package-json-removed.sh
+++ b/test/scripts/package-json-removed.sh
@@ -1,0 +1,18 @@
+#!/bin/sh
+
+# create a repo where a package.json is added in a branch
+
+# setup new git repo
+git init --initial-branch=main
+# needed for git run in CI context where none of this is set
+git config --local user.email "you@example.com"
+git config --local user.name "Your Name"
+
+echo '{ "name": "test-project", "dependencies": { "debug": "4.3.4"} }' > package.json
+git add package.json
+git commit -a -m "add package.json"
+
+# add package.json in new branch
+git checkout -b branch
+git rm package.json
+git commit -a -m "remove package.json"

